jsm.go

```diff
@@ -58,9 +58,12 @@ type JetStreamManager interface {
 	// The stream must have been created/updated with the AllowDirect boolean.
 	GetLastMsg(name, subject string, opts ...JSOpt) (*RawStreamMsg, error)
 
-	// DeleteMsg erases a message from a stream.
+	// DeleteMsg deletes a message from a stream. The message is marked as erased, but its value is not overwritten.
 	DeleteMsg(name string, seq uint64, opts ...JSOpt) error
 
+	// SecureDeleteMsg deletes a message from a stream. The message value is overwritten with random data.
+	SecureDeleteMsg(name string, seq uint64, opts ...JSOpt) error
```

Comment:
I thought about that, but I feel that in this case (having really only 2 options), 2 separate methods are a bit more straightforward to the user.  `JSOpt` is already confusingly large bag of options and finding the one particular option which is applicable for a specific method is a hard job, even with good documentation. 

Providing 2 methods on the other hand, should eliminate any confusion. If a user wants to delete message, they will stumble on 2 methods with relevant names: `DeleteMsg()`, which looks like a good starting point and `SecureDeleteMsg()`. Both have clear documentation and are not configurable (which is a good thing IMO).

WDYT?

Reply:
I started with adding a new API for the "direct get" but was told that it was better to add options instead. I do agree - however - that the JSOpt can be a bit confusing.
To sump up, I don't have objections per-se to add a new API (actually, in the C client I have `js_DeleteMsg()` and `js_EraseMsg()` APIs), just felt that the will was to go with options instead of new APIs. If other reviewers in the list are OK with new API, I am ok too.